# STevoCRC: single-cell and spatial omics reveal spatiotemporal evolution of tumor heterogeneity and tumor microenvironment in colorectal cancer (CRC).
## Abstract
Abundant and diverse immune cells in intestinal organ are excluded from colorectal cancer (CRC), which is associated with a very low response rate to immune checkpoint blockade (ICB) in CRC. However, the detailed spatiotemporal development of this immunotherapy-resistance niche during CRC progression remains poorly understood. Here, we performed spatiotemporally resolved multi-omics analysis on adjacent mucosa, peripheral blood, polyps, primary and metastatic tumor tissues from 358 patients, including more than 1 million cells with single-cell RNA-sequencing, 316,603 spots with spatial transcriptomics, 174, 351, 171 samples with whole exome sequencing, bulk RNA-seq, and proteomics, respectively. We identified four distinct TME ecotypes from dynamic change of 82 cell types during CRC development, associated with eight meta-programs of tumor states. Notably, we characterized a spatiotemporal hostile niche, gradually formed by the aggregation of SPP1+ macrophages, CCL3+ neutrophil, FAP+ fibroblast, and tip cells around tumor boundary via a dynamic interaction with T lymphocytes as CRC progressed, which explains establishment of the “fried ice-cream” structure of CRC. An Apcmin/+/KrasG12D/G12D/Trp53-/-(AKPmut) organoid-derived mouse CRC model recapitulates the spatiotemporal evolution of tumor microenvironment that we observed in human and the depletion of one hostile cell type neutrophil in vivo managed to sensitize ICB-mediated immunotherapy in CRC. Our study reconstructs the spatiotemporal evolution of the CRC TME in silico and in vivo, highlighting the key role of stromal-immune crosstalk in this process and shedding light on potentially hostile niche targeting strategies to boost ICB mediated immunotherapy.

![image](https://github.com/Yelab2020/STevoCRC/assets/97086137/39435d15-bc57-4172-b65e-0ca9997e2e1b)
